content/blogs/2023-11-15-kestra-sqs.md

```diff
@@ -0,0 +1,233 @@
+
+To keep things simple, we'll just log the URI of the incoming SQS message.
+
+But first, you need to establish a connection to SQS. You'll have to specify the values for the AWS access key, AWS secret key, AWS region, and the queue URL. All of these were added to environment variables in the previous section, meaning you can access them with `"{{ secret('<VARIABLE-NAME>') }}"`. Add all of them to a new trigger of type `io.kestra.plugin.aws.sqs.Trigger`:
+
+```yaml
+id: sqs-trigger
+namespace: dev
+
+tasks:
+  - id: hello
+    type: io.kestra.core.tasks.log.Log
+    message: "{{ trigger.uri }}"
+
+triggers:
+  - id: sqs
+    type: io.kestra.plugin.aws.sqs.Trigger
+    accessKeyId: "{{ secret('AWS_ACCESS_KEY') }}"
+    secretKeyId: "{{ secret('AWS_SECRET_KEY') }}"
+    region: "{{ secret('AWS_REGION') }}"
+    queueUrl: "{{ secret('SQS_URL') }}"
+    maxRecords: 1
+```
+
+The `maxRecords` property set to 1 means one execution of everything specified under `tasks` will process exactly 1 SQS message.
+
+**There's no need to run this flow - just save it**. To test things out, we'll go over to AWS and send a new message to the queue manually:
+
+![Image 5 - Manually sending message to a queue](/blogs/2023-11-15-kestra-sqs/5.png)
+
+The `io.kestra.core.tasks.log.Log` plugin included in our flow logic was set to log the trigger URI. You can access all Kestra logs under "Logs" in the navigation menu:
+
+![Image 6 - Kestra logs](/blogs/2023-11-15-kestra-sqs/6.png)
+
+As you can see, the URI of the processed SQS message was logged successfully.
+
+You can click on the `executionId` value of this log to further inspect everything that happened behind the scenes: 
+
+![Image 7 - Execution details](/blogs/2023-11-15-kestra-sqs/7.png)
+
+And to finally verify the message you sent manually was retrieved by Kestra, you can preview or download the file at the logged URI. Here's what it contains:
+
+![Image 8 - Flow execution Gantt view](/blogs/2023-11-15-kestra-sqs/8.png)
+
+If you want a set-it-and-forget-it way of processing SQS messages with Kestra, this is the only approach you'll need. But if you want a bit more manual control, make sure to read the two upcoming examples.
+
+
+### Example 2: Push Messages to AWS SQS
+In this example, you'll write a Kestra flow that manually pushes a message to SQS.
+
+It's a good practice to create `taskDefaults` for `io.kestra.plugin.aws.sqs.Publish`. This will ensure we don't have to repeat ourselves and also will allow us to change credentials in one place.
+
+The `io.kestra.plugin.aws.sqs.Publish` plugin also accepts `from` parameter, which further expects one or more `data` values to be submitted to the queue. You can read more about message formatting on our [official documentation](https://kestra.io/plugins/plugin-aws/tasks/sqs/io.kestra.plugin.aws.sqs.publish).
+
+Our flow will separate the two message publishing tasks with a simple Python script task that sleeps for a couple of seconds:
```

Comment:
```suggestion
It's a good practice to create `taskDefaults` for `io.kestra.plugin.aws.sqs.Publish`. This will ensure we don't have to repeat ourselves and also will allow us to change credentials in one place.

The `io.kestra.plugin.aws.sqs.Publish` plugin also accepts `from` parameter, which further expects one or more `data` values to be submitted to the queue. You can read more about message formatting in our [official documentation](https://kestra.io/plugins/plugin-aws/tasks/sqs/io.kestra.plugin.aws.sqs.publish).

Our flow will separate the two message publishing tasks with a simple Python script task that sleeps for a couple of seconds:
```